node/src/components/rpc_server.rs

```diff
@@ -352,56 +369,79 @@ where
+                async move {
+                    responder
+                        .respond(
+                            effect_builder
+                                .get_available_block_range_from_storage()
+                                .await,
+                        )
+                        .await
+                }
+                .ignore()
             }
-            .ignore(),
             Event::RpcRequest(RpcRequest::SpeculativeDeployExecute {
                 block_header,
                 deploy,
                 responder,
-            }) => self.handle_execute_deploy(effect_builder, block_header, *deploy, responder),
+            }) if self.speculative_exec_enable => {
+                self.handle_execute_deploy(effect_builder, block_header, *deploy, responder)
+            }
             Event::GetBlockResult {
                 maybe_id: _,
                 result,
                 main_responder,
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::GetBlockTransfersResult {
                 result,
                 main_responder,
                 ..
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::QueryGlobalStateResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::QueryEraValidatorsResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetBidsResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetBalanceResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetDeployResult {
                 hash: _,
                 result,
                 main_responder,
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::GetPeersResult {
                 peers,
                 main_responder,
-            } => main_responder.respond(peers).ignore(),
+            } if self.enable => main_responder.respond(peers).ignore(),
+            Event::RpcRequest(RpcRequest::SpeculativeDeployExecute { .. })
+                if self.speculative_exec_enable =>
+            {
+                Effects::new()
+            }
```

Comment:
What's happening here? Why do we ignore the request even if the endpoint is enabled.

Reply:
Fixed.